src/istio/chart/templates/gateway.yaml

```diff
@@ -27,6 +27,7 @@ spec:
         {{- range $server.hosts | default (list "*") }}
         - "{{ . }}.{{ $domain }}"
         {{- end }}
+      {{- if $.Values.tls.enabled }}
```

Comment:
Definitely see the utility here but would be very hesitant to ship this as-is without some safeguards to ensure that TLS is being enforced at a different layer (i.e. using those AWS LB annotations). I don't have any immediate great ideas here but there might be some ways to approach this using...
1. Zarf actions to check something `onDeploy.before`?
2. Helm jobs to check something pre-install/pre-upgrade?
3. Pepr Policy to block this gateway config except in x/y/z scenario?

The Pepr Policy is probably the most "uds friendly" approach, followed maybe by the zarf action. Regardless of approach we should also document this configuration/how it should be used in our ingress doc I think.

Reply:
After working through some Pepr Policy stuff at first, it seems like the more straightforward way to implement some safeguards is adding a Helm job/chart hook that checks for the annotations prior to installing the gateway object. I've added this in my latest commit.

Reply:
We don't have to go the pepr policy route if it seems too challenging, but just as a note for reference - we generate all the types used by Pepr using [this KFC command](https://github.com/defenseunicorns/kubernetes-fluent-client?tab=readme-ov-file#generating-typescript-definitions-from-crds). It should be pretty easy to get the necessary gateway type added if desired. For example, the prometheus CRDs got added with this:
```console
npx kubernetes-fluent-client crd https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.76.1/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml src/pepr/operator/crd/generated/prometheus
```

Reply:
I decided to go back and work the Pepr Policy solution instead. Much cleaner and better UX!

<img width="831" alt="Screenshot 2024-11-22 at 12 24 56 PM" src="https://github.com/user-attachments/assets/0d09aa75-6583-473b-ae1b-ede628a64395">